openshift-kube-apiserver/admission/customresourcevalidation/apiserver/validate_apiserver.go

```diff
@@ -200,7 +200,9 @@ func validateTLSSecurityProfileType(fieldPath *field.Path, profile *configv1.TLS
 			errs = append(errs, field.Required(fieldPath.Child("intermediate"), fmt.Sprintf(typeProfileMismatchFmt, profile.Type)))
 		}
 	case configv1.TLSProfileModernType:
-		errs = append(errs, field.NotSupported(fieldPath.Child("type"), profile.Type, availableTypes))
+		if profile.Modern == nil {
+			errs = append(errs, field.Required(fieldPath.Child("modern"), fmt.Sprintf(typeProfileMismatchFmt, profile.Type)))
```

Comment:
given that the TLS profile is set for many components, should we add a test in origin to ensure the platform supports it?

Reply:
Okay, I think we should add a test first, ensure the entire platform runs on the new profile, and only then merge this PR.